### ISSUE TITLE
复现结果与论文相差约10%，是否有训练细节遗漏？

### ISSUE THREAD

**Issue by LINSJGM** [Scu-sen/HiCervix#2]
Hi maintainers,

我在复现论文实验时发现模型在测试集上的准确率比论文中报告的结果低了约 10%。我使用了仓库提供的配置文件和训练流程进行了完整的训练和评估，但最终性能仍存在一定差距。

我的精度为
Level_1 accuracy: 0.8537
level_1 hierarchical distanse: 0.1014
##############################
Level_2 accuracy: 0.6982
level_2 hierarchical distande: 0.2691
##############################
Level_3 accuracy: 0.6965
level_3 hierarchical distande: 0.3114
##############################
Average accuracy: 0.7495
Average hierarchical distance: 0.2273

### 训练配置如下：

{   'alpha': 0.4,
    'arch': 'swinT',
    'barzdenzler': False,
    'batch_size': 8,
    'beta': 0,
    'data': 'tiered-imagenet-224',
    'data_augmentation': True,
    'data_dir': '../data/',
    'data_path': None,
    'data_paths_config': '../data_paths.yml',
    'devise': False,
    'devise_single_negative': False,
    'dropout': 0.5,
    'epochs': 50,
    'expm_id': '',
    'fc_inner_dim': 1024,
    'gpu': 0,
    'log_freq': 100,
    'loss': 'hierarchical-cross-entropy',
    'lr': 1e-05,
    'lr_fc': 0.001,
    'num_training_steps': 200000,
    'optimizer': 'adam_amsgrad',
    'out_folder': 'hierswin_alpha0.4',
    'output': 'hierswin_alpha0.4',
    'pretrained': True,
    'pretrained_folder': None,
    'seed': None,
    'shuffle_classes': False,
    'soft_labels': False,
    'start_epoch': 0,
    'target_size': 384,
    'train_backbone_after': inf,
    'train_csv': train_hierswin.csv',
    'use_2fc': False,
    'use_fc_batchnorm': False,
    'val_csv': val_hierswin.csv',
    'val_freq': 1,
    'weight_decay': 0.0,
    'weight_decay_fc': 0.0,
    'workers': 2}

希望可以得到一些帮助或建议，以缩小与论文结果之间的差距。如果有任何我没有注意到的细节，也欢迎指出！

Thanks in advance! @Xiyue-Wang @Scu-sen @DeCai01

**Comment by DeCai01** [comment]
Thank you for your clarification and for highlighting this important point. We appreciate the opportunity to address this misunderstanding and provide a detailed explanation of our methodology.

**​​Key Clarifications​​**
​​Pretrained Model Source​​
We utilized the ​​[official Swin Transformer checkpoint​​](https://github.com/SwinTransformer/storage/releases/download/v1.0.0/swin_large_patch4_window12_384_22k.pth) released on GitHub  to initialize training for our HiCervix dataset. 
By the way, we provide  the finetuned Swin Transformer checkpoint​​ on HiCervix dataset which can be [downloaded](https://drive.google.com/file/d/1TsULQjIsLHsKnO6DOin7MNXQy5Ycw3M9/view?usp=drive_link) here.
**​​Training Script Modifications​​**
Below is the revised command with key adjustments for reproducibility:
```
cd HierSwin  

##  Execute training with specified parameters  
python3 scripts/start_training.py \
    --arch swinT \                  # Architecture: Swin Transformer  
    --loss hierarchical-cross-entropy \  # Custom hierarchical loss for multi-level classification  
    --pretrained_folder "pretrained_Swin_Transformer/*.pth" \  # Path to pretrained weights  
    --alpha 0.4 \                  # Weighting factor for hierarchical loss (balances global/local features)  
    --output hierswin_alpha0.4 \   # Output directory for logs and checkpoints  
    --train-csv train_hierswin.csv \  # Training data CSV with hierarchical labels  
    --val-csv val_hierswin.csv \    # Validation data CSV  
    --epochs 50                    # Training epochs
```
